ledger/src/shred.rs

```diff
@@ -389,17 +391,23 @@ impl Shred {
     dispatch!(pub fn payload(&self) -> &Payload);
     dispatch!(pub fn sanitize(&self) -> Result<(), Error>);
 
-    // Only for tests.
-    dispatch!(pub fn set_index(&mut self, index: u32));
-    dispatch!(pub fn set_slot(&mut self, slot: Slot));
-
     pub fn copy_to_packet(&self, packet: &mut Packet) {
         let payload = self.payload();
         let size = payload.len();
         packet.buffer_mut()[..size].copy_from_slice(&payload[..]);
         packet.meta_mut().size = size;
     }
 
+    #[cfg(feature = "dev-context-only-utils")]
+    pub fn set_slot(&mut self, slot: u64) {
+        self.common_header_mut().slot = slot;
+    }
+
+    #[cfg(feature = "dev-context-only-utils")]
+    pub fn set_index(&mut self, index: u32) {
+        self.common_header_mut().index = index;
+    }
```

Comment:
`set_{index,slot}` were intentionally not implemented for Merkle shreds.
These methods break shreds integrity and make it invalid.

Please remove all instances of `common_header_mut`.

Reply:
Yes but they are also critical for every second unittest. I am considering disabling integrity checks vs improving our ability to produce invalid shreds in unittests.